### ISSUE TITLE
Publish objects in frame of supporting object

### ISSUE THREAD

**Issue by MatthijsBurgh** [tue-robotics/ed#58]
This would reduce the displacement of objects when a supporting object is moved. Thus reduce errors in association of objects.
@reinzor 
@jlunenburg 
@rokusottervanger 
What do you think?

**Comment by reinzor** [comment]
Everything is stored w.r.t. the map frame so I don't know what you mean with "publish". How to publish if this information is not present? I do think it is more logical to store entities w.r.t. their supporting objects but the current implementation of ED does not allow this. Properly implementing relative positions is going to be a lot of work and requires a lot of re-implementation of functionalities. We have done this in the past so if you are interested in doing this work, I can point you in the right direction. There are already pose crawlers etc. implemented.

**Comment by MatthijsBurgh** [comment]
I do mean store. Let's have a short talk about this tomorrow. Then we can decide if this could be a side project.

**Comment by rokusottervanger** [comment]
I agree with the idea. It feels more 'correct' to do something like that. The thought crossed my mind a few times as well. Let's talk about it tomorrow.

**Comment by jlunenburg** [comment]
In principle, this seems to make (much more) sense. Two comments:
- didn't we want to get rid of the huge amount of published tfs? In that case, we would have to cook up a better solution instead of simply attaching a different frame id
- practically: this will probably break a lot of stuff. If we want to do this, it might not be the right time.

Anyway, didn't hear this discussion yesterday but might be good to do this in the near future

**Comment by rokusottervanger** [comment]
Question remains: is there an issue here? It's a lot of work just to check if it improves anything. So we should know that it is worth the effort. Also, there are other bugs in the association. I'm pretty sure of that. Maybe we should fix those first?

**Comment by MatthijsBurgh** [comment]
I already labeled the issue as an enhancement. I do have seen some problems with the association during testing last week. But I don't have the knowledge to know what is the low hanging fruit.
